cmd/pbm-agent/profile.go

```diff
@@ -0,0 +1,209 @@
+		l.Error("missed config profile name")
+		return
+	}
+
+	ctx = log.SetLogEventToContext(ctx, l)
+
+	var err error
+	defer func() {
+		if err != nil {
+			l.Error("failed to add config profile: %v", err)
+		}
+	}()
+
+	nodeInfo, err := topo.GetNodeInfo(ctx, a.nodeConn)
+	if err != nil {
+		err = errors.Wrap(err, "get node info")
+		return
+	}
+	if !nodeInfo.IsLeader() {
+		l.Debug("not leader. skip")
+		return
+	}
+
+	lck := lock.NewLock(a.leadConn, lock.LockHeader{
+		Type:    ctrl.CmdAddConfigProfile,
+		Replset: a.brief.SetName,
+		Node:    a.brief.Me,
+		OPID:    opid.String(),
+		Epoch:   util.Ref(epoch.TS()),
+	})
+
+	got, err := a.acquireLock(ctx, lck, l)
+	if err != nil {
+		l.Error("acquiring lock: %v", err)
+		return
+	}
+	if !got {
+		l.Error("lock not acquired")
+		return
+	}
+	defer func() {
+		l.Debug("releasing lock")
+		err = lck.Release()
+		if err != nil {
+			l.Error("unable to release lock %v: %v", lck, err)
+		}
+	}()
+
+	err = cmd.Storage.Cast()
+	if err != nil {
+		l.Error("storage cast: %v", err)
+		return
+	}
+
+	stg, err := util.StorageFromConfig(&cmd.Storage, log.LogEventFromContext(ctx))
+	if err != nil {
+		err = errors.Wrap(err, "storage from config")
+		return
+	}
+
+	err = storage.HasReadAccess(ctx, stg)
+	if err != nil {
+		if !errors.Is(err, storage.ErrUninitialized) {
+			l.Error("check read access: %v", err)
+			return
+		}
+
+		err = storage.Initialize(ctx, stg)
+		if err != nil {
+			l.Error("init storage: %v", err)
+			return
+		}
+	}
+
+	profile := &config.Config{
+		Name:      cmd.Name,
+		IsProfile: true,
+		Storage:   cmd.Storage,
+	}
+	err = config.AddProfile(ctx, a.leadConn, profile)
+	if err != nil {
+		l.Error("add profile config: %v", err)
+		return
+	}
+
+	l.Info("profile saved")
+}
+
+func (a *Agent) handleRemoveConfigProfile(
+	ctx context.Context,
+	cmd *ctrl.ProfileCmd,
+	opid ctrl.OPID,
+	epoch config.Epoch,
+) {
+	logger := log.FromContext(ctx)
+
+	if cmd == nil {
+		l := logger.NewEvent(string(ctrl.CmdRemoveConfigProfile), "", opid.String(), epoch.TS())
+		l.Error("missed command")
+		return
+	}
+
+	l := logger.NewEvent(string(ctrl.CmdRemoveConfigProfile), cmd.Name, opid.String(), epoch.TS())
+	if cmd.Name == "" {
+		l.Error("missed config profile name")
+		return
+	}
+
+	ctx = log.SetLogEventToContext(ctx, l)
+
+	var err error
+	defer func() {
+		if err != nil {
+			l.Error("failed to remove config profile: %v", err)
+		}
+	}()
+
+	nodeInfo, err := topo.GetNodeInfo(ctx, a.nodeConn)
+	if err != nil {
+		err = errors.Wrap(err, "get node info")
+		return
+	}
+	if !nodeInfo.IsLeader() {
+		l.Debug("not leader. skip")
+		return
+	}
+
+	lck := lock.NewLock(a.leadConn, lock.LockHeader{
+		Type:    ctrl.CmdRemoveConfigProfile,
+		Replset: a.brief.SetName,
+		Node:    a.brief.Me,
+		OPID:    opid.String(),
+		Epoch:   util.Ref(epoch.TS()),
+	})
+
+	got, err := a.acquireLock(ctx, lck, l)
+	if err != nil {
+		l.Error("acquiring lock: %v", err)
+		return
+	}
+	if !got {
+		l.Error("lock not acquired")
```

Comment:
This command should be run on a specific node (so-called backup leader). This is an error because the command, basically, cannot be done. Other commands put DEBUG level because they could be proceeded by another agent.

In this command, it is better to use ERROR level. For other, we can review what is better.

Reply:
> logging manager solution will be in 'red' all the time

we should expose errors. users on their side could configure a filter to reduce "noise". but we shouldn't hide an error if we cannot continue with it.